app/controllers/users_controller.rb

```diff
@@ -0,0 +1,34 @@
+class UsersController < ApplicationController
+
+  def new
+    @user = User.new
+    if request.xhr?
+      render :new, layout: false
+    else
```

Comment:
What if it's not xhr? If you just do the default you don't need the else clause.